pkg/operator/baremetal_config.go

```diff
@@ -69,12 +70,19 @@ func getBaremetalProvisioningConfig(dc dynamic.Interface, configName string) (Ba
 		glog.Errorf("provisioningDHCPRange not found in Baremetal provisioning CR %s", configName)
 		return BaremetalProvisioningConfig{}, err
 	}
+	provisioningOSDownloadURL, found, err := unstructured.NestedString(provisioningSpec, "provisioningOSDownloadURL")
+	if !found {
```

Comment:
shouldn't you  be also checking `err != nil` in all this statements?

Reply:
We are propagating the error up to the caller. The caller can decide what to do with it. At this time, we are calling it from syncBaremetalControllers(). We are ignoring the error there (just logging it) and proceeding with the deployment because we have the option of reading the config from the configmap.
We are in the processing of getting to the point where we will not need the configmap altogether. There will be a PR that will make updates to this logic where we will remove all references to the configmap and then an error while reading the provisioning CR will be acted upon.
The changes in this PR relate to only 1 topic of adding OS Download URL to CR and associated changes to read it. In the interest of not bunching up a whole lot of changes in one PR, I'll propose a new PR to read only the CR and not the configmap.

Reply:
this is propagating the error up only when !found. Shouldn't this be doing `if !found || err != nil {}`

Reply:
Ok. Let me make that change as you requested.

I looked at https://github.com/kubernetes/apimachinery/blob/f30c0235171023905e0c3cf646c7f16409384495/pkg/apis/meta/v1/unstructured/helpers.go#L53 which is called from https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/unstructured/helpers.go#L74. There is no case when found is true and then there is an error. I understand that this underlying implementation could change so we should be checking for both perhaps.

Reply:
Made the necessary update.